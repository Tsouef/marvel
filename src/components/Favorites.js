import React, { Component } from 'react';
import { connect } from 'react-redux';

class Favorites extends Component {
  renderContent() {
    return this.props.favorites.map((favorite, index) => {
      const { thumbnail, name, description } = favorite;
      return (
        <div className="character">
          <div className="character-cover">
            <img alt="" src={`${thumbnail.path}.${thumbnail.extension}`} />
          </div>
          <div className="character-description">
            <p>
              <span className="meta">Name: </span> {name}
            </p>
            <p>
              <span className="meta">Description: </span>
              <span dangerouslySetInnerHTML={{ __html: description }} />
            </p>
          </div>
        </div>
      );
    });
  }

  render() {
    return <div>{this.renderContent()}</div>;
  }
}

function mapStateToProps({ favorites }) {
  return { favorites };
}

export default connect(mapStateToProps, null)(Favorites);
