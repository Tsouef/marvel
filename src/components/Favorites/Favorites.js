import React, { Component } from 'react';
import { connect } from 'react-redux';

import { deleteFavorite } from '../../actions';

export class Favorites extends Component {
  renderContent() {
    return this.props.favorites.favorites.map((favorite, index) => {
      const { thumbnail, name, description, id } = favorite;
      return (
        <div key={id} className="card">
          <div className="card-image">
            <img alt="" src={`${thumbnail.path}.${thumbnail.extension}`} />
            <a
              onClick={() => this.props.deleteFavorite(id)}
              className="btn-floating halfway-fab waves-effect waves-light red"
            >
              <i className="material-icons">delete</i>
            </a>
          </div>
          <div className="card-content">
            <span className="card-title">{name}</span>
            <p>
              <span dangerouslySetInnerHTML={{ __html: description }} />
            </p>
          </div>
        </div>
      );
    });
  }

  render() {
    if (this.props.favorites.favorites.length <= 0) {
      return <div>Choose some favorites</div>;
    }
    return <div>{this.renderContent()}</div>;
  }
}

function mapStateToProps({ favorites }) {
  return { favorites };
}

export default connect(mapStateToProps, { deleteFavorite })(Favorites);
