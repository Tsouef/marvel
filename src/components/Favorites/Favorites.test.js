import React from 'react';
import renderer from 'react-test-renderer';

import { Favorites } from './Favorites';

describe('Favorites Component', () => {
  it('should render Favorites', () => {
    const props = {
      favorites: {
        favorites: []
      }
    };

    const tree = renderer.create(<Favorites {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Favorites with elements', () => {
    const props = {
      favorites: {
        favorites: [
          {
            thumbnail: 'thumbnail',
            name: 'name',
            description: 'description',
            id: 1
          }
        ]
      }
    };

    const tree = renderer.create(<Favorites {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('should delete a favorite', () => {
  //   const deleteFavorite = jest.fn();
  //   const props = {
  //     deleteFavorite: jest.fn(),
  //     favorites: {
  //       favorites: [
  //         {
  //           thumbnail: 'thumbnail',
  //           name: 'name',
  //           description: 'description',
  //           id: 1
  //         }
  //       ]
  //     }
  //   };
  //   const wrapper = shallow(<Favorites {...props} />);
  //   const instance = wrapper.dive().instance();
  //
  //   console.log(instance);
  //   instance().deleteFavorite(1);
  // });
});
