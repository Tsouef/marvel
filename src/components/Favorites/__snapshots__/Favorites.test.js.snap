// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Component should render Favorites 1`] = `
<div>
  Choose some favorites
</div>
`;

exports[`Favorites Component should render Favorites with elements 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt=""
        src="undefined.undefined"
      />
      <a
        className="btn-floating halfway-fab waves-effect waves-light red"
        onClick={[Function]}
      >
        <i
          className="material-icons"
        >
          delete
        </i>
      </a>
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title"
      >
        name
      </span>
      <p>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "description",
            }
          }
        />
      </p>
    </div>
  </div>
</div>
`;
